plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -8837,10 +8853,23 @@ public void testTimestampWithTimeZone()
                         OptionalLong.empty()),
                 PrincipalPrivileges.fromHivePrivilegeInfos(Collections.emptySet()));
 
-        assertThat(query("SELECT * FROM test_timestamptz"))
-                .matches("VALUES TIMESTAMP '2022-07-26 17:13:00.000 UTC'");
+        long microsFraction = switch (timestampPrecision) {
+            case MILLISECONDS -> 123;
+            case MICROSECONDS, NANOSECONDS -> 123456;
+        };
+        long nanosFraction = switch (timestampPrecision) {
+            case MILLISECONDS -> 123;
+            case MICROSECONDS -> 123457;
+            case NANOSECONDS -> 123456789;
+        };
+        assertThat(query(session, "SELECT * FROM test_timestamptz"))
+                .matches("VALUES " +
+                        "(TIMESTAMP '2022-07-26 17:13:14.123 UTC')," +
+                        "(TIMESTAMP '2022-07-26 17:13:14." + microsFraction + " UTC')," +
+                        "(TIMESTAMP '2022-07-26 17:13:14." + nanosFraction + " UTC')");
 
         assertUpdate("DROP TABLE test_timestamptz");
+        assertUpdate("DROP TABLE test_timestamptz_base");
```

Comment:
We could add a similar test in Iceberg connector. e.g. 
```java
    @Test // TestIcebergV2 
    public void testInt96TimestampWithTimeZone()
    {
        assertUpdate("CREATE TABLE hive.tpch.test_timestamptz_base (t timestamp) WITH (format = 'PARQUET')");
        assertUpdate("INSERT INTO hive.tpch.test_timestamptz_base (t) VALUES" +
                "(timestamp '2022-07-26 12:13')", 1);

        // Writing TIMESTAMP WITH LOCAL TIME ZONE is not supported, so we first create Parquet object by writing unzoned
        // timestamp (which is converted to UTC using default timezone) and then creating another table that reads from the same file.
        String tableLocation = metastore.getTable("tpch", "test_timestamptz_base").orElseThrow().getStorage().getLocation();

        // TIMESTAMP WITH LOCAL TIME ZONE is not mapped to any Trino type, so we need to create the metastore entry manually
        metastore.createTable(
                new io.trino.metastore.Table(
                        "tpch",
                        "test_timestamptz",
                        Optional.of("hive"),
                        "EXTERNAL_TABLE",
                        new Storage(
                                HiveStorageFormat.PARQUET.toStorageFormat(),
                                Optional.of(tableLocation),
                                Optional.empty(),
                                false,
                                Collections.emptyMap()),
                        List.of(new Column("t", HiveType.HIVE_TIMESTAMPLOCALTZ, Optional.empty(), Map.of())),
                        List.of(),
                        Collections.emptyMap(),
                        Optional.empty(),
                        Optional.empty(),
                        OptionalLong.empty()),
                PrincipalPrivileges.fromHivePrivilegeInfos(Collections.emptySet()));

        assertThat(query("SELECT * FROM hive.tpch.test_timestamptz"))
                .matches("VALUES TIMESTAMP '2022-07-26 17:13:00.000 UTC'");

        String path = (String) computeScalar("SELECT \"$path\" FROM hive.tpch.test_timestamptz_base");
        long size = (long) computeScalar("SELECT \"$file_size\" FROM hive.tpch.test_timestamptz_base");

        // Read a Parquet file from Iceberg connector
        assertUpdate("CREATE TABLE iceberg.tpch.test_timestamptz_migrated(t TIMESTAMP(6) WITH TIME ZONE)");

        BaseTable table = loadTable("test_timestamptz_migrated");

        table.updateProperties()
                .set(DEFAULT_NAME_MAPPING, toJson(MappingUtil.create(table.schema())))
                .commit();

        table.newAppend()
                .appendFile(DataFiles.builder(table.spec())
                        .withPath(path)
                        .withFormat(PARQUET)
                        .withFileSizeInBytes(size)
                        .withRecordCount(1)
                        .build())
                .commit();

        assertThat(query("SELECT * FROM iceberg.tpch.test_timestamptz_migrated"))
                .matches("VALUES TIMESTAMP '2022-07-26 17:13:00.000000 UTC'");

        assertUpdate("DROP TABLE hive.tpch.test_timestamptz_base");
        assertUpdate("DROP TABLE hive.tpch.test_timestamptz");
        assertUpdate("DROP TABLE iceberg.tpch.test_timestamptz_migrated");
    }
```

Reply:
added the above test, thanks

Reply:
I guess this test should be enough and we don't need test in `TestHiveSparkCompatibility`